### ISSUE TITLE
**isSpace** is not defined

### ISSUE THREAD

**Issue by marvin-season** [markdown-it/markdown-it#1082]
Function **isSpace** is not defined, may be the isSpace is not build into dist (esm)

const tokens = new MarkdownIt().parse('1.hh') // error isSpace is not defined

switch markdown v13 is okay

**Comment by mark-addams001881** [comment]
Hello, we tried to solve the issue.

This is what we did:

Export the isSpace function from utils.mjs and import it in parser_inline.mjs to make it available in the ESM build.

You can review changes in this commit: https://github.com/mark-addams001881/markdown-it-markdown-it-1082/commit/c218202c3a1bec66dfd9b93331a03968c427cb25.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by jonathanong** [comment]
i am getting this with nextjs when building with turbo pack

```js
 ⨯ ReferenceError: isSpace is not defined
    at __TURBOPACK__default__export__ (utils/markdown.mjs:9:35)
    at <unknown> (app/<redacted>)
    at Array.map (<anonymous>)
    at __TURBOPACK__default__export__ (<redacted>)
   7 | })
   8 |
>  9 | export default (text) => text ? md.render(text) : ''
     |                                   ^
  10 | {
  digest: '3474557066'
}
```

works fine without turbo pack

**Comment by Tharak-N** [comment]
I also face the same issue when using with next.js with turbopack, is there any other solution to fix this ?

**Comment by dragosrotaru** [comment]
I just changed my import to a require() and it worked

**Comment by daifuyang** [comment]
same as me in nextjs 15

**Comment by wjsbgsnwss** [comment]
Change to a lower version (npm install --save-dev markdown-it@13.0.2) and the issue is disappeared.

**Comment by ifeora-emeka** [comment]
Same issue here NextJs 15 Turbo.

has anyone been able to find a solution?

**Comment by TextC0de** [comment]
Same issue

**Comment by toryshadow** [comment]
Same issue, latest nextJs with turbo. Any fix for this?

**Comment by nazebzurati** [comment]
Possible workaround that works for me: disable turbopack OR use the Markdown component as client component with `use client;` directives. I'm using Next.js 15.2.2.

**Comment by dominicOT** [comment]
> Change to a lower version (npm install --save-dev markdown-it@13.0.2) and the issue is disappeared.

are you sure this works..? cos I have experienced lots of issues with changing libraries to their lower version

**Comment by Frnt-End** [comment]
> > Change to a lower version (npm install --save-dev markdown-it@13.0.2) and the issue is disappeared.
> 
> are you sure this works..? cos I have experienced lots of issues with changing libraries to their lower version

It helps me and fix the problem.

**Comment by dominicOT** [comment]
> > > Change to a lower version (npm install --save-dev markdown-it@13.0.2) and the issue is disappeared.
> > 
> > 
> > are you sure this works..? cos I have experienced lots of issues with changing libraries to their lower version
> 
> It helps me and fix the problem in my Next.js 15 project.

yeah, I did the downgrade already to a lower version and it works

**Comment by prajjwal2-3** [comment]
@tmacc man you're my hero ❤️

**Comment by Cygra** [comment]
For those who would like to avoid downgrading or are looking for a long-term fix, please follow this issue for updates and discussion: vercel/next.js#76802

**Comment by swampskye** [comment]
> Change to a lower version (npm install --save-dev markdown-it@13.0.2) and the issue is disappeared.

Works for me. Thanks!

**Comment by Sriraj-dev** [comment]
@tmacc  Man, You saved my Night! ❤️😂

**Comment by 1693943116** [comment]
are you ok

**Comment by dcolley** [comment]
In next.js I used this at the top of my page.tsx:
```ts
// Define isSpace function globally for markdown-it
declare global {
  function isSpace(code: number): boolean;
}

// Define the isSpace function
globalThis.isSpace = function(code: number): boolean {
  return code === 0x20 || code === 0x09 || code === 0x0A || code === 0x0B || code === 0x0C || code === 0x0D;
};
```

**Comment by puzrin** [comment]
Closing as reported to be solved https://github.com/markdown-it/markdown-it/issues/1082#issuecomment-3204882232

Note. The difference between v13 and v14: migrating from ancient `browserify` to `rollup`. The reported problem was related to Nuxt's tupbopack bug, not to this package, IMHO.
